### ISSUE TITLE
Slow response

### ISSUE THREAD

**Issue by RWOverdijk** [balderdashy/sails#2370]
Greetings,

I've noticed that sails seems to become quite slow when it has to return a bit more data than usual.

Some info:

* This is on the latest stable
* Latest sails-mysql
* Latest waterline (of the regular branch, not waterline2)
* I've checked the query waterline creates, and I've optimised it beyond belief (0.00s query)
* It does have some associations.

Example row:

```
*************************** 30. row ***************************
                            body: Ohhh,hey hey heyy! ahah :P
                            read: 1
                         initial: 0
                              id: 5956
                       createdAt: 2014-11-03 07:00:37
                       updatedAt: 2014-11-04 10:14:02
                          thread: 850
                            from: 3792
                              to: 14750
                thread___subject: As promised
                     thread___id: 850
              thread___createdAt: 2014-10-30 10:05:45
              thread___updatedAt: 2014-11-04 11:08:53
                   thread___from: 14750
                     thread___to: 3792
                 from___username: assepoesje
                 from___password: yyy
                    from___email: yyy@yyy.com
        from___notificationEmail: NULL
from___notificationEmailVerified: NULL
                  from___country: NULL
            from___emailVerified: 0
                 from___mailable: 1
                 from___socketId: NULL
              from___partnerCode: 61
              from___partnerInfo: typein
                    from___roles: ["performer"]
                       from___id: 3792
                from___createdAt: 2014-10-07 05:59:13
                from___updatedAt: 2014-11-04 12:11:06
                   from___object: 1
                  from___visitor: NULL
                from___performer: 3792
                   to___username: RWOverdijk
                   to___password: xxx
                      to___email: xxx@xxx.nl
          to___notificationEmail: NULL
  to___notificationEmailVerified: NULL
                    to___country: NL
              to___emailVerified: 1
                   to___mailable: 1
                   to___socketId: NULL
                to___partnerCode: 61
                to___partnerInfo: NULL
                      to___roles: ["visitor"]
                         to___id: 14750
                  to___createdAt: 2014-09-17 07:52:07
                  to___updatedAt: 2014-11-04 10:14:04
                     to___object: 1
                    to___visitor: 4
                  to___performer: NULL
```

This times 30 causes a load of 100%, and a 500 / 600 ms response time. This is unacceptable for real word applications (like ours, which is currently running in production, smoothly).

Is this a known issue? Is there something I can do? Something we can do to optimise performance at all?

Thanks,

**Comment by sailsbot** [comment]
Thanks for posting, @RWOverdijk.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by RWOverdijk** [comment]
I haven't had feedback on this yet.

**Comment by mikermcneil** [comment]
Hey @RWOverdijk, if you're still seeing this, would you let me know?

**Comment by RWOverdijk** [comment]
@mikermcneil Will do. I think it's related to https://github.com/balderdashy/sails-mysql/issues/161
